packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm

```diff
@@ -243,7 +251,7 @@ - (void)boundsDidChange:(CGRect)newBounds
 
   if (_state != nullptr) {
     auto newState = ModalHostViewState{RCTSizeFromCGSize(newBounds.size)};
-    _state->updateState(std::move(newState));
+    _state->updateState(std::move(newState), EventQueue::UpdateMode::unstable_Immediate);
```

Comment:
I asked internally, and unfortunately we are still experimenting with this API. So we can't merge it as it is.

It is currently guarded behind the `enableImmediateUpdateModeForContentOffsetChanges` which is specific for ScrollView... Perhaps, we can create another feature flag `enableImmediateUpdateForModalDetents` for this.